mne/viz/topomap.py

```diff
@@ -3015,3 +3015,74 @@ def _set_adjacency(adjacency, both_nodes, value):
 
     with warnings.catch_warnings(record=True):
         adjacency[both_nodes, both_nodes[::-1]] = value
+
+
+@fill_doc
+def plot_regression_weights(model, ch_type=None, vmin=None, vmax=None,
+                            cmap=None, sensors=True, colorbar=True, res=64,
+                            size=1, cbar_fmt='%.2g', show=True,
+                            show_names=False, title=None, mask=None,
+                            mask_params=None, outlines='head', axes=None,
+                            sphere=None, border=_BORDER_DEFAULT):
+    """Plot the regression weights of a fitted EOGRegression model.
+
+    Parameters
+    ----------
+    model : EOGRegression
+        The fitted EOGRegression model whose weights will be plotted.
+    %(ch_type_topomap)s
+    %(vmin_vmax_topomap)s
+    %(cmap_topomap)s
+    %(sensors_topomap)s
+    %(colorbar_topomap)s
+    %(res_topomap)s
+    %(size_topomap)s
+    %(cbar_fmt_topomap)s
+    %(show)s
+    %(show_names_topomap)s
+    %(title_none)s
+    %(mask_evoked_topomap)s
+    %(mask_params_topomap)s
+    %(outlines_topomap)s
+    %(axes_evoked_plot_topomap)s
+    %(sphere_topomap_auto)s
+    %(border_topomap)s
+
+    Returns
+    -------
+    fig : instance of matplotlib.figure.Figure
+        Figure with a topomap subplot for each channel type.
+
+    Notes
+    -----
+    .. versionadded:: 1.2
+    """
+    import matplotlib.pyplot as plt
+    sphere = _check_sphere(sphere)
+    ch_types = _get_channel_types(model.info, unique=True, only_data_chs=True)
+
+    nrows = model.coef_.shape[1]
+    ncols = len(ch_types)
+    fig, axes = plt.subplots(nrows, ncols, squeeze=False,
+                             figsize=(ncols * 2, nrows * 1.5 + 1))
+    axes = iter(axes.T.ravel())
+
+    for ch_type in ch_types:
+        data_picks, pos, merge_channels, names, ch_type, sphere, clip_origin =\
+            _prepare_topomap_plot(model, ch_type=ch_type, sphere=sphere)
+        outlines = _make_head_outlines(sphere, pos, outlines='head',
+                                       clip_origin=clip_origin)
+        coef = model.coef_[data_picks]
+        for data, ch_name in zip(coef.T, model._picks_artifact):
+            if merge_channels:
+                data, names = _merge_ch_data(data, ch_type, names)
+            ax = next(axes)
+
+            _plot_topomap_multi_cbar(
+                data, pos, ax, title=f'{ch_type}/{ch_name}', vmin=vmin,
+                vmax=vmax, cmap=cmap, outlines=outlines, colorbar=colorbar,
+                unit='', cbar_fmt=cbar_fmt, sphere=sphere, ch_type=ch_type)
+    fig.subplots_adjust(top=0.88, bottom=0.06, left=0.025, right=0.911,
+                        hspace=0.2, wspace=0.5)
```

Comment:
setting this is fine as long as the fig and axes were created by us. If the user passed their own `axes` we should avoid subplots_adjust and the like (and probably avoid setting `fig.suptitles` too).  try a `git grep axes_was_none` to see ways we've done this before.